nnvm/python/nnvm/frontend/tensorflow.py

```diff
@@ -974,24 +985,28 @@ def from_tensorflow(self, graph):
                 # Pass the node name too in attr
                 attr["_node_name"] = node.name
 
-                #ToDo: Some of the tensorflow operators internaly maintain
-                #execution layers and its output name will the layer number along with
-                #graph node name.eg: Node name:- 'Model/RNN/cell_0/RnnCell', but the
-                #output name will be 'Model/RNN/cell_0/RnnCell:0'. In this case,
-                #the digit has to be ignored.
-                if ":" in node.input[0]:
-                    in_name, _ = node.input[0].split(':')
-                    node.input[0] = in_name
-
-                # Fill shapes for all inputs in a list
-                try:
-                    inputs = [self._nodes[i] for i in node.input]
-                    for i in node.input:
-                        input_shapes[self._nodes[i]] = self._output_shapes[i]
-                    attr['_input_shapes'] = input_shapes
-                except KeyError:
-                    # TODO: Need to find clean way to handle '^CheckNumerics'
-                    pass
+                inputs = []
+                for node_input_name in node.input:
+                    node_input_key = node_input_name.split(':')
+                    slot_num = 0
+                    if len(node_input_key) > 1:
+                        slot_num = int(node_input_key[1])
+                    node_input_key = node_input_key[0]
+
+                    try:
+                        try:
+                            input_sym = self._nodes[node_input_key].__getitem__(slot_num)
+                        except NNVMError:
+                            # TODO: Fancy node name invalid slot neglect
+                            input_sym = self._nodes[node_input_key].__getitem__(0)
+
+                        inputs.append(input_sym)
+                        input_shapes[input_sym] = self._output_shapes[
+                            node_input_key].__getitem__(slot_num)
+                        attr['_input_shapes'] = input_shapes
+                    except KeyError:
```

Comment:
This except is for the same statement below where the ```self._nodes``` doesn't have key ```node_input_key```
```input_sym = self._nodes[node_input_key].__getitem__(slot_num)```

Suggest to make one try block with multiple except instead of nested.

Reply:
as address the [review](https://github.com/dmlc/tvm/pull/1572#discussion_r208894072) comment, we need nested try block to handle fancy node name convention logic.

Reply:
It's the same statement which cause 2 exceptions in two different cases, one try on ```input_sym = self._nodes[node_input_key].__getitem__(slot_num)``` with two ```except``` should work. Pls check.

Reply:
First try block we are trying to access the slot base input and after caught NNVMError we are trying to access default(0) location input, in this case if we caught error should be thrown to user.